src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/common/stateful_init_graph.cpp

```diff
@@ -0,0 +1,312 @@
+        inferRequestRef = compiledModelRef.create_infer_request();
+    }
+
+    const ov::element::Type netPrc = ElementType::f32;
+    ov::InferRequest inferRequestRef;
+};
+
+// ReadValue Assign direct pair
+//
+//             input_1   input_2
+//                |        |
+//              Add_1     /
+//                \      /
+//                 MatMul
+//                   |
+//   input_0     ReadValue ..........
+//       \      /       \           .
+//         Add_0      Assign ........
+//          |
+//        Result
+
+class InitGraphStatefulModel : public InitGraphStatefulModelBase {
+public:
+    void SetUp() override {
+        targetDevice = utils::DEVICE_CPU;
+
+        std::vector<InputShape> inputShapes;
+        bool directPair;
+        std::tie(inputShapes, directPair) = this->GetParam();
+
+        init_input_shapes(inputShapes);
+        ov::ParameterVector input_params;
+        for (auto&& shape : inputDynamicShapes) {
+            input_params.push_back(std::make_shared<ov::op::v0::Parameter>(netPrc, shape));
+        }
+
+        input_params[0]->set_friendly_name("input_0");
+        input_params[1]->set_friendly_name("input_1");
+        input_params[2]->set_friendly_name("input_2");
+
+        // init_graph
+        auto add_1 =
+            std::make_shared<ov::op::v1::Add>(input_params[1], ov::op::v0::Constant::create(netPrc, {1}, {1.0f}));
+        add_1->set_friendly_name("init_graph/add_1");
+        auto mm_0 = std::make_shared<ov::op::v0::MatMul>(add_1, input_params[2]);
+        mm_0->set_friendly_name("init_graph/mm_0");
+
+        const std::string variable_name("var_direct_pair");
+        auto variable = std::make_shared<ov::op::util::Variable>(
+            ov::op::util::VariableInfo{{inputDynamicShapes[1][0], inputDynamicShapes[2][1]}, netPrc, variable_name});
+
+        auto read = std::make_shared<ov::op::v6::ReadValue>(mm_0, variable);
+        std::shared_ptr<ov::Node> add_0 = std::make_shared<ov::op::v1::Add>(input_params[0], read);
+        add_0->set_friendly_name("add_0");
+        auto assign = std::make_shared<ov::op::v6::Assign>(directPair ? read : add_0, variable);
+        auto res = std::make_shared<ov::op::v0::Result>(add_0);
+        function = std::make_shared<ov::Model>(ov::ResultVector({res}), ov::SinkVector({assign}), input_params);
+    }
+
+    void check_init_graph_node() override {
+        // Node with friendly name "init_graph/add_1" and init_graph/mm_0 should be moved into subgraph.
+        bool found_init_graph_add = false;
+        bool found_init_graph_mm = false;
+        for (auto node : compiledModel.get_runtime_model()->get_ops()) {
+            if (node->get_friendly_name() == "init_graph/add_1") {
+                found_init_graph_add = true;
+                break;
+            }
+            if (node->get_friendly_name() == "init_graph/mm_0") {
+                found_init_graph_mm = true;
+                break;
+            }
+        }
```

Comment:
Using the names may be not the most reliable approach to check the graph final topology.
Could you please enhance the checks with counting the specific node types. Please refer to:
https://github.com/openvinotoolkit/openvino/blob/6805e6229add575fdae0f4a5d90e70713924bbfe/src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/common/sdpa_group_beam_search.cpp#L202-L205

Reply:
Done.